Kernel/FileSystem/FileBackedFileSystem.cpp

```diff
@@ -225,23 +241,24 @@ void DiskBackedFS::flush_writes_impl()
         if (!entry.is_dirty)
             return;
         u32 base_offset = static_cast<u32>(entry.block_index) * static_cast<u32>(block_size());
-        device().write_raw(base_offset, block_size(), entry.data);
+        m_file_description->seek(base_offset, SEEK_SET);
+        file().write(*m_file_description, entry.data, block_size());
```

Comment:
Can't you call `FileDescription::write()` directly? And don't forget to check the write succeeded (and wrote this many bytes). Also, we should check that the file is seekable upfront (when constructing ourselves) — you cannot, for example, mount Ext2FS from a pipe :slightly_smiling_face:

Reply:
Bump

Reply:
Another bump? 🙂